docs/design-proposals/arch/daemonset.md

```diff
@@ -72,3 +88,10 @@ There are two significant required functionalities for kubelet, at high level:
 2. Retrieving supporting resources by TP origin.
    <p/>Secret and configmap resources each is managed via a local cached store being replenished by external API servers (TP's). Current store implementation simply derives TP by tenant name of the resource. To accomodate the DaemonSet design, the cache store should accept requests to get resource with specified TP origin; it should be able to keep system-tenanted secret or configmap without losing their TP origins; also it should use the resource-specific TP origin to replenish the staled resource instead of the simple tenant-name derived one.
    <p/>The specific detail the changes of store should be carefully considered before the kubelet module implementation.
+3. Keeping track of TP origin of service in Service lister, using services from proper TP when identifying services of the Pod.
```

Comment:
There are <b>three</b> significant required functionalities for kubelet?

How is this done in service support for scale out? I assume it is very similar. Can there a design doc that can be referred here?

Reply:
Line 87, segguested - > suggested

Reply:
Also, I don't think it is best to derive TP origin from tenant name. There is no such fixed configuration. It is better track the mapping in kubelet internal memory when the info was first received. Isn't this how it was implemented in service support for scale out?

Reply:
1- 3 is out-of-date. will reword it; will correct typos; 
2- Service resource will be mentioned in section of kube-proxy change;
3- yes the current implementation is indeed tracking the tenant's origin on first-time receipt. will re-phase to reduce the confusion.